comictaggerlib/coverimagewidget.py

```diff
@@ -177,9 +177,9 @@ def set_issue_id(self, issue_id: int) -> None:
             self.update_content()
             self.issue_id = issue_id
 
-            comic_vine = ComicVineTalker()
-            ComicVineTalker.url_fetch_complete = self.sig.emit_url
-            comic_vine.async_fetch_issue_cover_urls(self.issue_id)
+            sources_api = SourcesAPI()
```

Comment:
Apart from the async issues with this, here is a list of modules that call CoverImageWidget class:
autotagmatchwindow
autotagprogresswindow
issueselectionwindow
matchselectionwindow
pagebrowser
pagelisteditor
taggerwindow
volumeselectionwindow

Not all of those present a problem but pagelisteditor would require passing SourcesAPI in when its called, along with some others. The current coverimagewidget may have created the new object for the async reason but also for the PITA it would be to pass it in? Unless there's an alternative to make passing SourceAPI optional so those can not do it? Seems like a bad idea though. Could reduce it down and make the new object once if ignoring the async problem?

Reply:
That is a good point. pagebrowser is the only one that wouldn't need it at all.
There are two options:
1. Pass a single object through the entire application
1. Obtain all of the necessary information to create a similar object, eg any options which may impact how it retrieves images and where it stores them, and then pass all of those through to go and create a new object

Right now this is accomplished using class variables, most likely for the exact reason that it is a pain to pass `ComicVineTalker`, through to everything, one of the problems with class variables (and global variables) is that anyone can change it and no one can have their own version of it, which means that you have to know everywhere that that class is used and if changing it will break anything.

It is a pain but using class or global variables is usually equivalent to hiding all the dirty clothes under the bed, someone still has to take them out and wash them eventually.